### PR TITLE
Wip/add items to position target global

### PR DESCRIPTION
**Pull request by hendjoshsr71**
For review only

### CODE REVIEW

File: ArduCopter/mode_auto.cpp [hunk 1]
```diff
@@ -685,18 +685,32 @@ int32_t ModeAuto::wp_bearing() const
...
 {
     switch (_mode) {
     case SubMode::NAVGUIDED:
-        return copter.mode_guided.get_wp(destination);
+        return copter.mode_guided.get_target_info(type_mask, target, target_vel, target_accel);
     case SubMode::WP:
-        return wp_nav->get_oa_wp_destination(destination);
+        type_mask = POSITION_TARGET_TYPEMASK_VX_IGNORE | POSITION_TARGET_TYPEMASK_VY_IGNORE | POSITION_TARGET_TYPEMASK_VZ_IGNORE |
+                    POSITION_TARGET_TYPEMASK_AX_IGNORE | POSITION_TARGET_TYPEMASK_AY_IGNORE | POSITION_TARGET_TYPEMASK_AZ_IGNORE |
+                    POSITION_TARGET_TYPEMASK_YAW_IGNORE| POSITION_TARGET_TYPEMASK_YAW_RATE_IGNORE; // ignore everything except position
```
Comment (hendjoshsr71) [inline ArduCopter/mode_auto.cpp:709]: Clean all of these up with a struct or enum like below,

```suggestion
        type_mask = POSITION_TARGET_TYPEMASK_VELOCITY_IGNORE | POSITION_TARGET_TYPEMASK_ACCEL_IGNORE | POSITION_TARGET_TYPEMASK_YAW_ALL_IGNORE; // ignore everything except position
```

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 2]
```diff
@@ -4750,6 +4750,41 @@ void GCS_MAVLINK::send_set_position_target_global_int(uint8_t target_system, uin
             0,0);   // yaw, yaw_rate
 }
 
+void GCS_MAVLINK::send_position_target_global_int()
+{
+    uint16_t type_mask;
+    Location target;
+    Vector3f target_vel, target_accel;
+    if (!AP::vehicle()->get_target_info(type_mask, target, target_vel, target_accel)) {
```
Comment (peterbarker) [inline libraries/GCS_MAVLink/GCS_Common.cpp:4849]: Why go via vehicle here?

File: ArduCopter/GCS_Mavlink.h [hunk 3]
```diff
@@ -23,7 +23,6 @@ class GCS_MAVLINK_Copter : public GCS_MAVLINK
 
     MAV_RESULT _handle_command_preflight_calibration(const mavlink_command_long_t &packet) override;
 
-    void send_position_target_global_int() override;
     void send_position_target_local_ned() override;
```
Comment (peterbarker) [inline ArduCopter/GCS_Mavlink.h]: Pretty sure this is a hint that your new function has to have global in it somewhere.

Perhaps we should deal with both at the same time - are the vels and accels in the same frame?

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 2 above)
Comment (hendjoshsr71) [reply libraries/GCS_MAVLink/GCS_Common.cpp:4849]: I don't really have to, but there was the scripting only function which `get_target_info()` is similar to just with more info.  Here, https://github.com/ArduPilot/ardupilot/blob/da3ff2c142b1e3a65d66533f0133da1249450245/libraries/AP_Vehicle/AP_Vehicle.h#L191

File: ArduCopter/GCS_Mavlink.h (hunk 3 above)
Comment (hendjoshsr71) [reply ArduCopter/GCS_Mavlink.h]: Yes, moving local was my next step. That is why I left it the name  `get_target_info()` more generic. Dealing with the position in send_position_target_local_ned() will just need to 

Hmm on vels and accels one would hope so....?? they come into guided that way but I will check if they come out that way...

**Comment by hendjoshsr71** [comment]
PRed to main to repo

**Pull request closed without merging**